### PR TITLE
PR merge workflows

### PR DESCRIPTION
**Pull request by byrnHDF**
Merge the Test Express workflows into the PR CI
Split merge request triggers into autotools vs cmake

### CODE REVIEW

**Comment by jhendersonHDF** [comment]
Why merge the TestExpress workflows into the CI rather than leaving it as a daily action like before? If it's because they don't take much time, then we don't need a separate workflow to begin with. But the reason to have them as a daily action is because certain VFDs are definitely slow with a TestExpress setting of 0, so having it be a daily action makes it easy to isolate that testing.

**Comment by byrnHDF** [comment]
> But the reason to have them as a daily action is because certain VFDs are definitely slow with a TestExpress setting of 0, so having it be a daily action makes it easy to isolate that testing.

They can be easily called from a vfd workflow as a callable workflow. And we can add an optional parameter to set a vfd env variable.

File: .github/workflows/testxpr-auto.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: hdf5 TestExpress Autotools CI
+
+# Run TestExpress CI daily at 04:00 CDT (9:00 UTC)
```
Comment (jhendersonHDF) [inline .github/workflows/testxpr-auto.yml]: Not relevant anymore

File: .github/workflows/testxpr-cmake.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+name: hdf5 TestExpress CMake CI
+
+# Run TestExpress CI daily at 04:00 CDT (9:00 UTC)
```
Comment (jhendersonHDF) [inline .github/workflows/testxpr-cmake.yml]: Same here

**Review by jhendersonHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**